Classes/UVWelcomeViewController.m

```diff
@@ -280,60 +248,44 @@ - (void)logoTapped {
 #pragma mark ===== UISearchBarDelegate Methods =====
 
 - (BOOL)searchBarShouldBeginEditing:(UISearchBar *)searchBar {
-    [searchBar setShowsCancelButton:YES animated:YES];
-    _filter = IA_FILTER_ALL;
-    searchBar.showsScopeBar = YES;
-    searchBar.selectedScopeButtonIndex = 0;
-    [searchBar sizeToFit];
-    
-    if (FORMSHEET) {
-        _tableView.tableHeaderView = searchBar;
-        _searching = YES;
-        [_tableView reloadData];
-    } else {
-        [_searchController setActive:YES animated:YES];
-        _searchController.searchResultsTableView.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
-    }
+    _filter = _searchController.searchBar.selectedScopeButtonIndex;
     return YES;
 }
 
 - (void)searchBar:(UISearchBar *)searchBar selectedScopeButtonIndexDidChange:(NSInteger)selectedScope {
     _filter = searchBar.selectedScopeButtonIndex;
-    if (FORMSHEET) {
-        [_tableView reloadData];
-    } else {
-        [_searchController.searchResultsTableView reloadData];
-    }
+    // DDSearch
+    [self updateSearchResultsForSearchController:_searchController];
 }
 
-- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
-    _instantAnswerManager.searchText = searchBar.text;
-    [_instantAnswerManager search];
+- (void)searchBarCancelButtonClicked:(UISearchBar *)searchBar {
+    // DDSearch
+    _searchController.searchBar.selectedScopeButtonIndex = 0;
+    _searchController.searchBar.text = @"";
+    _instantAnswerManager.instantAnswers = [NSArray array];
+    [_tableView reloadData];
 }
 
-- (void)searchBarCancelButtonClicked:(UISearchBar *)searchBar {
-    _searchBar.showsScopeBar = NO;
-    if (FORMSHEET) {
-        [_searchBar setShowsCancelButton:NO animated:YES];
-        _searchBar.text = @"";
-        _instantAnswerManager.instantAnswers = [NSArray array];
-        [_searchBar resignFirstResponder];
-        _searching = NO;
-        [_searchBar sizeToFit];
-        _tableView.tableHeaderView = _searchBar;
-        [_tableView reloadData];
+#pragma mark ==== UISearchResultsUpdating Methods ====
+
+// DDSearch
+- (void)updateSearchResultsForSearchController:(UISearchController *)searchController {
+    _instantAnswerManager.searchText = searchController.searchBar.text;
+    [_instantAnswerManager search];
+    
+    if (_searchController.searchResultsController) {
+        UVWelcomeSearchResultsController *searchResultsTVC = (UVWelcomeSearchResultsController *)_searchController.searchResultsController;
+        searchResultsTVC.searchResults = self.searchResults;
+        [searchResultsTVC.tableView reloadData];
```

Comment:
This is where I would expect an `updateResults` method to be called, as suggested above.